### PR TITLE
feat/RWSVIEWER 16  metadata

### PR DESCRIPTION
**Pull request by petergoes**
- feat(config.graphql): Get metadata for layers
- feat(add-translation): Make translation cli more resilient
- feat(LayerChooser): Add layer chooser component
- refactor(Layers): Use layer chooser component over active layers
- refactor: Embed LayerInfoDialog as slot in LayerControl
- refactor(store/map): addRasterLayer handles arrays
- chore: Remove obsolete LayerTree component
- refactor(config.graphql): Rename properties in metadata
- feat(translations): Add translations
- chode: Add schema files for config and search for future reference
- chore: Remove obsolete LayerList component

### CODE REVIEW

File: src/components/LayerChooser/LayerChooser.vue [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+    />
+          
+    <div class="layer-chooser__filter-options">
+      <v-switch
+        v-model="onlyActive"
+        class="layer-chooser__only-active-switch"
+        :label="$t('onlyActiveLayers')"
+      />
+      <!-- Discuss if this is a useful feature -->
+      <!-- <v-btn
```
Comment (vvandijck) [inline src/components/LayerChooser/LayerChooser.vue:19]: Yes, could be useful, let's discuss with Gerrit

File: src/components/LayerChooser/LayerChooser.vue [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+    </v-treeview>
+  </v-sheet>
+</template>
+
+<script>
+  import { difference } from 'ramda'
+  import { mapGetters, mapActions } from 'vuex'
+  import LayerInfoDialog from '../LayerInfoDialog/LayerInfoDialog'
+  import addParentIdToLayers from '~/lib/add-parent-id-to-layers'
+  const LayerControl = () => import('~/components/LayerControl/LayerControl.vue')
```
Comment (vvandijck) [inline src/components/LayerChooser/LayerChooser.vue:104]: `.vue` is probably not necessary.

File: src/components/LayerChooser/LayerChooser.vue [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+  align-items: center;
+  justify-content: flex-end;
+  gap: $spacing-tiny;
+}
+
+.layer-chooser__only-active-switch {
+  margin-right: auto !important;
+}
+
+.layer-chooser__tree {
```
Comment (vvandijck) [inline src/components/LayerChooser/LayerChooser.vue:190]: Empty rule can be removed.

File: src/components/LayerControl/LayerControl.vue [hunk 4]
```diff
@@ -34,6 +40,8 @@
 </template>
 
 <script>
+  
+
```
Comment (vvandijck) [inline src/components/LayerControl/LayerControl.vue:44]: Remove extra newlines.

File: src/components/LayerChooser/LayerChooser.vue [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+        </layer-control>
+      </template>
+    </v-treeview>
+  </v-sheet>
+</template>
+
+<script>
+  import { difference } from 'ramda'
+  import { mapGetters, mapActions } from 'vuex'
+  import LayerInfoDialog from '../LayerInfoDialog/LayerInfoDialog'
```
Comment (vvandijck) [inline src/components/LayerChooser/LayerChooser.vue:102]: Use `~/components/` for consistency.

File: src/components/LayerInfoDialog/LayerInfoDialog.vue [hunk 6]
```diff
@@ -22,39 +22,53 @@
...
+
+<style lang="scss">
+.layer-info-dialog__metadata {
+  display: grid;
+  grid-template-columns: auto 1fr;
+  row-gap: $spacing-smaller;
+  column-gap: $spacing-default;
+}
+
+.layer-info-dialog__metadata div {
```
Comment (vvandijck) [inline src/components/LayerInfoDialog/LayerInfoDialog.vue:71]: I would not style directly on elements, using a class is more future-proof.

**Review by vvandijck — CHANGES_REQUESTED** [review]
Overall a very nice addition with lots of other improvements. The code is very clear. I only found a few small things. There's one issue (see screen recording below) we can take a look at together. I managed to fix this while working on it a few weeks ago.

https://user-images.githubusercontent.com/12752012/146170352-fe9f07da-5505-45ca-ab92-4bedf3acc51d.mov

File: src/components/LayerInfoDialog/LayerInfoDialog.vue (hunk 6 above)
Comment (petergoes) [reply src/components/LayerInfoDialog/LayerInfoDialog.vue:71]: Only a `<div>` is allowed next to `<dt>`, and `<dd>` elements in an `<dl>`. There may not be other elements, so the `<div>` is not going to change. The `<div>` only serves as a tool for the `:key` prop. The content of this rule complements that (using `display: contents` makes it practically unusable for other styling. Adding a class can give the reader of the markup the idea that the div does more than 'just exist'

Comment (vvandijck) [reply src/components/LayerInfoDialog/LayerInfoDialog.vue:71]: Ok yes that makes sense 👍🏼

**Review by vvandijck — APPROVED** [review]
✅

**Pull request merged**